src/features/MainPage/index.js

```diff
@@ -1,53 +1,35 @@
 import { nanoid } from "nanoid";
-import { useDispatch, useSelector } from "react-redux";
-import { useEffect } from "react";
-import { Container, MoviePageLink, LoadingPage, ErrorPage } from "./styled";
+import { useSelector, useDispatch } from "react-redux";
+import { Container, MoviePageLink } from "./styled";
 import { TilesContainer } from "../../common/Tiles/MovieTilesContainer/styled";
 import { TilesHeader } from "../../common/Tiles/TilesHeader/styled";
 import { MovieTile } from "../../common/Tiles/MovieTilesContainer/MovieTile";
 import Pagination from "../../common/Pagination";
-import { selectCurrentPage } from "../../common/Pagination/paginationSlice";
-import {
-  fetchCurrentMoviesPage,
-  selectPopularMoviesData,
-  selectPopularMoviesStatus,
-} from "./popularMoviesSlice";
-import {
-  fetchMovieTypesInit,
-  selectMovieTypesData,
-} from "../../common/movieTypes/movieTypesSlice";
+import { selectPopularMoviesStatus } from "./popularMoviesSlice";
 import noMovieImage from "./no-movie-image.svg";
 import { fetchMovieId } from "../MoviePage/movieDetailsSlice";
+import { usePopularMoviesData } from "./usePopularMoviesData";
+import { useCurrentPage } from "./useCurrentPage";
+import { useMovieTypesData } from "./useMovieTypesData";
+import { toMainPage } from "../../core/routes";
 
-function MainPage() {
-  const popularMoviesData = useSelector(selectPopularMoviesData);
-  const popularMoviesStatus = useSelector(selectPopularMoviesStatus);
-  const movieTypesData = useSelector(selectMovieTypesData);
-  const currentPage = useSelector(selectCurrentPage);
-
-  const imageBaseUrl = "https://image.tmdb.org/t/p/w300";
-  const numberOfMovieTypes = 3;
-
+const MainPage = () => {
   const dispatch = useDispatch();
-
-  useEffect(() => {
-    dispatch(fetchCurrentMoviesPage(currentPage));
-  }, [currentPage, dispatch]);
-
-  useEffect(() => {
-    dispatch(fetchMovieTypesInit());
-  }, [dispatch]);
+  const popularMoviesStatus = useSelector(selectPopularMoviesStatus);
+  const popularMoviesData = usePopularMoviesData();
+  const { movieTypesData, numberOfMovieTypes } = useMovieTypesData();
+  const currentPage = useCurrentPage();
 
   return (
     <>
       {popularMoviesStatus === "loading" ? (
-        <LoadingPage />
```

Comment:
I think that LoadingPage is necessary in MainPage, isn't it?

Reply:
Yes, it is. I left doing LoadingPage for later. In my opinion, the best solution for this will be adding placeholders